File: backend_fastapi/app/routes/conversations.py
from fastapi import APIRouter, HTTPException
from app.storage.file_store import get_all_conversations, get_conversation_by_id


router = APIRouter()


@router.get("/")
async def list_conversations():
    items = get_all_conversations(limit=50)
    return {"items": items}


@router.get("/{conversation_id}")
async def get_conversation(conversation_id: str):
    item = get_conversation_by_id(conversation_id)
    if item is None:
        raise HTTPException(status_code=404, detail={"error": "Not found"})
    return item

File: backend/src/routes/conversations.ts
import { Router, Request, Response } from 'express';
import { getAllConversations, getConversationById } from '../store/fileStore';

export const conversationRouter = Router();

// GET /conversations
conversationRouter.get('/', async (_req: Request, res: Response) => {
  const items = await getAllConversations({ limit: 50 });
  res.json({ items });
});

// GET /conversations/:conversationId
conversationRouter.get('/:conversationId', async (req: Request, res: Response) => {
  const { conversationId } = req.params;
  const item = await getConversationById(conversationId);
  if (!item) {
    return res.status(404).json({ error: 'Not found' });
  }
  return res.json(item);
});


